### ISSUE TITLE
"Parsing error: Unexpected token" when invoking anonymous arrow-functions

### ISSUE THREAD

**Issue by namuol** [babel/babel-eslint#263]
The parser fails when invoking arrow functions inline, like so:

```js
const fourtyTwo = () => { return 42 }()
//                                   ^ Unexpected token
```

Minimal reproduction:

```js
()=>{}()
//    ^ Unexpected token
```

Workarounds:

- Use `function` instead:

```js
const fourtyTwo = function () { return 42 }()
// This works fine
```

- Wrap the arrow function with parens:

```js
const fourtyTwo = (() => { return 42 })()
```

**Comment by hzoo** [comment]
If you ever get Unexpected token then it's an issue with babylon the parser (either unsupported or a syntax error)

In this case, I don't think that is valid so you would have to wrap it in parens like you added above. You can check on http://astexplorer.net/ for the other parsers as well
